src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -1352,10 +1352,12 @@ OGRLayerH QgsOgrProviderUtils::setSubsetString( OGRLayerH layer, GDALDatasetH ds
 {
   // Remove any comments
   QStringList lines {subsetString.split( QChar( '\n' ) )};
+
   lines.erase( std::remove_if( lines.begin(), lines.end(), []( const QString & line )
   {
     return line.startsWith( QLatin1String( "--" ) );
   } ), lines.end() );
+
```

Comment:
Just because it jumped my eyes, can you add a `as_const` and `const QString`, in the for loop below?

Reply:
Can't be const: line is altered a few lines below.